File: medplum-typescript/packages/sdk/src/model/fhir/OrganizationAffiliation.ts
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { ContactPoint } from "./ContactPoint";
import { Extension } from "./Extension";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { Period } from "./Period";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface OrganizationAffiliation {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier[] | null | undefined;
    active: boolean | null | undefined;
    period: Period | null | undefined;
    organization: Reference | null | undefined;
    participatingOrganization: Reference | null | undefined;
    network: Reference[] | null | undefined;
    code: CodeableConcept[] | null | undefined;
    specialty: CodeableConcept[] | null | undefined;
    location: Reference[] | null | undefined;
    healthcareService: Reference[] | null | undefined;
    telecom: ContactPoint[] | null | undefined;
    endpoint: Reference[] | null | undefined;
}

File: medplum-java/medplum-model/src/main/java/com/fhir/OrganizationAffiliation.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Boolean;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableOrganizationAffiliation.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface OrganizationAffiliation {
  Optional<List<ResourceList>> contained();

  Optional<Period> period();

  String resourceType();

  Optional<List<Reference>> location();

  Optional<List<ContactPoint>> telecom();

  Optional<List<Reference>> network();

  Optional<List<CodeableConcept>> specialty();

  Optional<List<Reference>> endpoint();

  Optional<List<Extension>> extension();

  Optional<Reference> organization();

  Optional<List<Reference>> healthcareService();

  Optional<code> language();

  Optional<List<CodeableConcept>> code();

  Optional<Boolean> active();

  Optional<Meta> meta();

  Optional<List<Extension>> modifierExtension();

  Optional<uri> implicitRules();

  Optional<Narrative> text();

  Optional<id> id();

  Optional<Reference> participatingOrganization();

  Optional<List<Identifier>> identifier();

  static ImmutableOrganizationAffiliation.ResourceTypeBuildStage builder() {
    return ImmutableOrganizationAffiliation.builder();
  }
}
